packages/calypso-e2e/src/lib/pages/full-site-editor-page.ts

```diff
@@ -210,6 +243,26 @@ export class FullSiteEditorPage {
 		await inserter.selectBlockInserterResult( blockName );
 	}
 
+	/**
+	 * Shared submethod to wait for a just-added block and get its block ID.
+	 * The ID is the most reliable way to identify the block over time.
+	 *
+	 * @param {string} blockEditorSelector Selector to find the block once added.
+	 * @returns The ID of the recently added block.
+	 */
+	private async getIdOfAddedBlock( blockEditorSelector: string ): Promise< string > {
+		// The added block will always either be focused, or will be the parent of a focused block.
+		const addedBlockLocator = this.editorCanvas.locator(
+			`${ selectors.focusedBlock( blockEditorSelector ) },${ selectors.parentOfFocusedBlock(
+				blockEditorSelector
+			) }`
+		);
+		await addedBlockLocator.waitFor();
+
+		return await getIdFromBlock( addedBlockLocator );
+	}
+
+	// TODO: Add this to the Gutenberg component to make it re-usable across editors!
```

Comment:
These TODOs are just ideas I had while I was in here, so adding them for future me :)